### PR TITLE
histogram: in hover labels, harmonize start/end values of shared bins

### LINKED ISSUE #4648: stacked histograms have different start/end values on hover
https://codepen.io/nicolaskruchten/pen/BaNxYzM

This is weird:

image

### PR DESCRIPTION
**Pull request by antoinerg**
Fixes #4648 

Codepen: https://codepen.io/antoinerg/pen/LYVrJoZ

cc @nicolaskruchten | @plotly/plotly_js

### CODE REVIEW

File: src/traces/histogram/calc.js [hunk 1]
```diff
@@ -124,10 +132,17 @@ function calc(gd, trace) {
...
+    roundFnOpts.leftGap = leftGap;
+    roundFnOpts.rightGap = rightGap;
 
     var roundFn;
     if(!uniqueValsPerBin) {
-        roundFn = getBinSpanLabelRound(leftGap, rightGap, binEdges, pa, calendar);
+        roundFn = function(v, isRightEdge) {
+            return function() {
+                var roundFnOpts = gd._fullLayout._roundFnOpts[groupName];
+                return getBinSpanLabelRound(roundFnOpts.leftGap, roundFnOpts.rightGap, binEdges, pa, calendar)(v, isRightEdge);
```
Comment (archmoj) [inline src/traces/histogram/calc.js]: This line is getting little too long. Maybe:
```js
return getBinSpanLabelRound(
  roundFnOpts.leftGap, 
  roundFnOpts.rightGap, 
  binEdges, pa, calendar
)(v, isRightEdge);
```

File: src/traces/histogram/calc.js [hunk 2]
```diff
@@ -102,12 +102,20 @@ function calc(gd, trace) {
         };
     }
 
+    if(!gd._fullLayout._roundFnOpts) gd._fullLayout._roundFnOpts = {};
+    var groupName = trace['_' + mainData + 'bingroup'];
+    var roundFnOpts = {leftGap: Infinity, rightGap: Infinity};
+    if(groupName) {
+        if(!gd._fullLayout._roundFnOpts[groupName]) gd._fullLayout._roundFnOpts[groupName] = roundFnOpts;
+        roundFnOpts = gd._fullLayout._roundFnOpts[groupName];
```
Comment (archmoj) [inline src/traces/histogram/calc.js]: Non-blocking, 
We could rewrite this:
```js
if(!gd._fullLayout._roundFnOpts[groupName]) {
  gd._fullLayout._roundFnOpts[groupName] = roundFnOpts;
}
roundFnOpts = gd._fullLayout._roundFnOpts[groupName];
```
to make it easier to read 3 `gd._fullLayout._roundFnOpts[groupName]`.

**Comment by archmoj** [comment]
This branch is little behind the master.
To help Q&A it may be a good idea to merge the master into this branch.

**Comment by archmoj** [comment]
Screenshot from 2020-03-18 20-43-44

File: src/traces/histogram/calc.js (hunk 1 above)
Comment (antoinerg) [reply src/traces/histogram/calc.js]: Done in c387f9b74050731396044ec1bd381073fdf0799e

**Comment by antoinerg** [comment]
> This branch is little behind the master.
> To help Q&A it may be a good idea to merge the master into this branch.

Is there anything that landed on `master` that is related to histogram?

**Pull request merged**